### PR TITLE
Improve Update Workflow testing UX

### LINKED ISSUE #1639: Improve UX for testing Workflow Update with the Workflow Test Environment
**Is your feature request related to a problem? Please describe.**
Improve UX for testing Workflow Update with the Workflow Test Environment. Currently users using the TestEnviorment have to implement the `UpdateCallbacks` to assert the result of an update call

**Describe the solution you'd like**
Provide helpers in the SDK for common update testing scenarios

**Additional context**
https://github.com/temporalio/samples-go/blob/main/reqrespupdate/workflow_test.go#L40

### PR DESCRIPTION
**Pull request by yuandrew**
## What was changed

Added default implementation `TestUpdateCallbacks` for basic test scenarios

## Why?

so tests don't need to implement `UpdateCallbacks` themselves. To make testing a little easier

## Checklist

1. Closes #1639

2. How was this tested:

existing unit tests that were effected still pass

### CODE REVIEW

**Comment by yuandrew** [comment]
@Quinn-With-Two-Ns  lmk if there's other improvements you were thinking about when you filed the issue that I didn't cover

File: internal/workflow_testsuite.go [hunk 1]
```diff
@@ -83,6 +84,15 @@ type (
 		runFn        func(args mock.Arguments)
 		waitDuration func() time.Duration
 	}
+
+	// TestUpdateCallback is a basic implementation of the UpdateCallbacks interface for testing purposes.
+	// Tests are welcome to implement their own version of this interface if they need to test more complex
+	// update logic. This is a simple implementation to make testing basic Workflow Updates easier.
+	TestUpdateCallback struct {
```
Comment (Quinn-With-Two-Ns) [inline internal/workflow_testsuite.go:91]: The issue was intended to improve the UX for users so we would need to expose this https://github.com/temporalio/sdk-go/blob/master/testsuite/testsuite.go#L35

Comment (yuandrew) [reply internal/workflow_testsuite.go:91]: oops! didn't realize this was needed. Thanks for pointing it out

File: internal/workflow_testsuite.go [hunk 2]
```diff
@@ -83,6 +84,15 @@ type (
...
+		accept   func()
```
Comment (Quinn-With-Two-Ns) [inline internal/workflow_testsuite.go:92]: These fields are private so you also need to make these public so the type can be constructed. To make them public you need to capitalize the first letter. I would also suggest prefixing them with `On` to be a bit more consistent with the other test environment API.

Comment (yuandrew) [reply internal/workflow_testsuite.go]: Thanks :) tested with https://github.com/temporalio/samples-go/pull/371, hopefully not missing anything else

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Pull request merged**